### DISCUSSION TITLE
Cannot find context for 'fork'

### DISCUSSION THREAD

**Discussion by MastracciN** [coqui-ai/TTS#1834] [General Q&A]
Hey everyone, I have been trying to train a model using the VITS recipe and LJSpeech. I have been encountering a ValueError where context cannot be found for 'fork'. I have tried changing the code in the trainer.py file. Some sources online recommended using 'spawn' instead but I found the program would fail after completing training. I'm currently running on a Windows system and tried using Cygwin to launch the program as well but still received the same errors. Any thoughts would be deeply appreciated

![image](https://user-images.githubusercontent.com/38020285/183124454-5bd0c82e-ca1c-4a1c-a892-8f7137501044.png)
Above error is received when multiprocesssing is set to 'fork'

![image](https://user-images.githubusercontent.com/38020285/183124709-2cdf113c-e5f3-413d-b472-e9f0fcac4c43.png)
Above error is received when multiprocessing is set to 'spawn'

**Comment by erogol** [comment]
I remember it got fixed when I call the training code under `if __name__ == "__main__"`

**Reply by MastracciN** [reply]
I may have misunderstood where to use it. I tried it here and was met with a similar PermissionError to what was posted above. I tried this in the trainer.py class. 
![image](https://user-images.githubusercontent.com/38020285/183671172-9fab6552-e435-43a8-9066-247aa6383f6b.png)
I also tried wrapping the entire train_vits.py recipe inside of it and received the same error
![image](https://user-images.githubusercontent.com/38020285/183726920-40157643-c5e5-412c-965f-5b03fbdf0074.png)
This is the full error if it provides any insight

**Reply by erogol** [reply]
I've no idea. Win is a different animal for me.
